utils/create_dummy_models.py

```diff
@@ -0,0 +1,741 @@
+import argparse
+import collections.abc
+import importlib
+import inspect
+import json
+import os
+import shutil
+import sys
+from pathlib import Path
+
+from datasets import load_dataset
+
+from check_config_docstrings import get_checkpoint_from_config_class
+from transformers import (
+    CONFIG_MAPPING,
+    FEATURE_EXTRACTOR_MAPPING,
+    PROCESSOR_MAPPING,
+    TOKENIZER_MAPPING,
+    AutoTokenizer,
+    LayoutLMv3TokenizerFast,
+    PreTrainedTokenizerFast,
+    logging,
+)
+from transformers.feature_extraction_utils import FeatureExtractionMixin
+from transformers.file_utils import is_tf_available, is_torch_available
+from transformers.models.auto.configuration_auto import AutoConfig, model_type_to_module_name
+from transformers.processing_utils import ProcessorMixin, transformers_module
+from transformers.tokenization_utils_base import PreTrainedTokenizerBase
+
+
+logging.set_verbosity_error()
+logger = logging.get_logger(__name__)
+
+sys.path.append(".")
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
+
+if not is_torch_available():
+    raise ValueError("Please install PyTorch.")
+
+if not is_tf_available():
+    raise ValueError("Please install TensorFlow.")
+
+FRAMEWORKS = ["pytorch", "tensorflow"]
+INVALID_ARCH = []
+TARGET_VOCAB_SIZE = 1024
+
+_pytorch_arch_mappings = [
+    x
+    for x in dir(transformers_module)
+    if x.startswith("MODEL_") and x.endswith("_MAPPING") and x != "MODEL_NAMES_MAPPING"
+]
+_tensorflow_arch_mappings = [
+    x for x in dir(transformers_module) if x.startswith("TF_MODEL_") and x.endswith("_MAPPING")
+]
+# _flax_arch_mappings = [x for x in dir(transformers_module) if x.startswith("FLAX_MODEL_") and x.endswith("_MAPPING")]
+
+pytorch_arch_mappings = [getattr(transformers_module, x) for x in _pytorch_arch_mappings]
+tensorflow_arch_mappings = [getattr(transformers_module, x) for x in _tensorflow_arch_mappings]
+# flax_arch_mappings = [getattr(transformers_module, x) for x in _flax_arch_mappings]
+
+unexportable_model_architectures = []
+
+ds = load_dataset("wikitext", "wikitext-2-raw-v1")
+training_ds = ds["train"]
+testing_ds = ds["test"]
```

Comment:
All of this should happen under the `__main__` method IMO; right now importing stuff from this script results in this code being run, which isn't ideal

Reply:
Could do it. However, I am not sure if we would import anything from this script (and not running the tiny model creation).